internal/graph/check.go

```diff
@@ -640,13 +641,37 @@ func (c *LocalChecker) buildCheckAssociatedObjects(req *ResolveCheckRequest, obj
 			},
 		}
 		objectType, relation := tuple.SplitObjectRelation(objectRel)
+
+		directlyRelatedUsersetTypes, err := typesys.DirectlyRelatedUsersets(objectType, relation)
+		if err != nil {
+			return nil, err
+		}
+
+		user := reqTupleKey.GetUser()
+		userType := strings.SplitN(user, ":", 2)
+
+		hasPubliclyAssignedType := false
+		// TODO: memorize the check for public wildcard
+		for _, directlyRelatedUsersetType := range directlyRelatedUsersetTypes {
+			if directlyRelatedUsersetType.GetWildcard() != nil && directlyRelatedUsersetType.GetType() == userType[0] {
+				hasPubliclyAssignedType = true
+			}
+		}
+
+		userFilter := []*openfgav1.ObjectRelation{{
+			Object: reqTupleKey.GetUser(),
+		}}
+		if hasPubliclyAssignedType {
```

Comment:
Isn't the other path faster 🤔

Reply:
No - the other path requires O(N) query where N is number of member to the group
This is O(1) query - we simply add the user:* as filter